src/WebJobs.Script/Workers/SharedMemoryDataTransfer/MemoryMappedFileAccessor.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.IO.MemoryMappedFiles;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Workers.SharedMemoryDataTransfer
+{
+    /// <summary>
+    /// Encapsulates functionality for accessing <see cref="MemoryMappedFile"/>.
+    /// There are platform specific implementations of this:
+    /// 1) <see cref="MemoryMappedFileAccessorWindows"/>
+    /// 2) <see cref="MemoryMappedFileAccessorLinux"/>
+    /// </summary>
+    public abstract class MemoryMappedFileAccessor : IMemoryMappedFileAccessor
+    {
+        public MemoryMappedFileAccessor(ILogger<MemoryMappedFileAccessor> logger)
+        {
+            Logger = logger;
+        }
+
+        protected ILogger Logger { get; }
+
+        public abstract bool TryCreate(string mapName, long size, out MemoryMappedFile mmf);
```

Comment:
Review the accessibility of these methods. Seems like they should all be protected, since they're implementation details not intended to be called externally?

Reply:
They are being used in the unit tests.

Reply:
Internal methods can be accessed in unit tests.

Reply:
But these are part of the `IMemoryMappedFileAccessor` interface - hence they need to be public, right?